### PR TITLE
Added Stacks.c

### LINKED ISSUE #1018: Stack using linkedlist in c
Your repository already has stack using linked list in cpp,python and java. I want to do it in C.

### PR DESCRIPTION
**Pull request by divya1999ann**
Fixes #1018 

Checklist:
 - [x] 4 space indentation.
 - [x] Coding conventions are followed.
 - [x]  Sample Input / Output.
 - [x]  Logic Documentation (Comments).
 - [x]  File names are correct.

Changes proposed in this pull request:
- Adding Stack using LinkedList in C

Languages Used:
- C

Files Added:
- Stacks.c

### CODE REVIEW

**Comment by somya-kapoor** [comment]
Hey, The code looks good and the topic you've chosen is nice. But, while going through it, I found exact similarity of your code with the code available at https://gist.github.com/dineshrajpurohit/0bb67d29a039f85f2f10 . Please, re-code it yourself and then it would be good.@jainaman224

**Comment by divya1999ann** [comment]
I have done the same in c. Please go through it.

**Comment by marwahmanbir** [comment]
@divya1999ann mention the issue number and add the sections that you've removed from the PR template above & complete them.

**Comment by divya1999ann** [comment]
Yes I have completed the template.

**Comment by marwahmanbir** [comment]
Also complete the checklist

**Comment by divya1999ann** [comment]
Completed the checklist.

**Comment by marwahmanbir** [comment]
@divya1999ann you've mentioned the PR number instead of issue number in the template above.

**Comment by divya1999ann** [comment]
where do I find the issue number.

**Comment by divya1999ann** [comment]
@marwahmanbir where can I find the issue number.

**Comment by marwahmanbir** [comment]
Did you open an issue before opening this PR? You have to open one, and get it assigned by the admin or a mentor before you can start working on the issue.

**Comment by marwahmanbir** [comment]
If you did create an issue you can find it at the issues section of this repository (https://github.com/jainaman224/Algo_Ds_Notes/issues)

**Comment by divya1999ann** [comment]
@marwahmanbir I have opened an issue but nobody has repsoneded to it.Anyways I have corrected the issue number in the template.

**Comment by somya-kapoor** [comment]
@divya1999ann Please use 4 space indentation instead of a tab.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 1]
```diff
@@ -10,7 +10,7 @@ int data[100];
 /*Function to push element to stack */
 void PUSH(struct stack *s,int value,int n)
 {
-	if(s->top==n-1)
+    	if(s->top==n-1)
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Please add space around ==.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/* C program to implement stack using linked list*/
+
+#include<stdio.h>
+struct stack
+{
+int top;
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Please fix indentation

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+int data[100];
+}Stack; 
+
+/*Function to push element to stack */
+void PUSH(struct stack *s,int value,int n)
+{
+    	if(s->top==n-1)
+		{
+		 printf("\nStack is full");
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Please fix the indentation.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+		}
+	else 
+	{s->top++;
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Please fix indentation

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+	 s->data[s->top]=value;
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Please add space around =

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	}
+}
+
+/*Function to pop element from stack */
+void POP(struct stack *s)
+{
+	if(s->top==-1)
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Add space

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+	 printf("\nThe stack is empty");
+	else
+	 {int value;
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Fix indentation

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+	  value=s->data[s->top];
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Add space

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+void main()
+{
+	struct stack *s;
+	s=&Stack;
+	Stack.top=-1;
+	int c,value,k,n;
+	printf("\nEnter the number of elements");
+	scanf("%d",&n);
+	do
+	   {
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: fix indentation

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+	  value=s->data[s->top];
+	  printf("\nThe poped value is %d",value);
+	  s->top--;
+	 }
+}
+/* Main function */
+void main()
+{
+	struct stack *s;
+	s=&Stack;
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Add space

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+	Stack.top=-1;
+	int c,value,k,n;
```
Comment (somya-kapoor) [inline Stacks_Using_Linked_Lists/Stacks.c]: Add space after ,

**Comment by divya1999ann** [comment]
I have made the changes . Please go through it @somya-kapoor

File: Stacks_Using_Linked_Lists/Stacks.c (hunk 2 above)
Comment (somya-kapoor) [reply Stacks_Using_Linked_Lists/Stacks.c]: struct stack
{
      int top;
      int data[100];
}Stack;

**Comment by somya-kapoor** [comment]
@divya1999ann Please fix the indentation. 
for example, see https://user-images.githubusercontent.com/42409030/56458160-d884a680-63a0-11e9-90db-e26a3bdafd73.png.

**Comment by divya1999ann** [comment]
@somya-kapoor I have made the changes.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+{
+int top;
+int data[100];
+}Stack; 
+
+/*Function to push element to stack */
+void PUSH(struct stack *s,int value,int n)
+{
+    if(s->top == n-1)
+        {
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: Please fix indentation in all if-else blocks.

The block from line 69 to 78 looks good. Please follow that style everywhere.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+            printf("\nStack is full");
+        }
+    else 
+        {
+	    s->top++;
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: This line is indented with tab. Please use 4 space indentation

**Comment by SakshiShreya** [comment]
Please also add these functions-

1. Peek
2. Is Empty
3. Size

**Comment by jainaman224** [comment]
@somya-kapoor @KanchanThareja Please review.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            int value;
+            value = s->data[s->top];
+            printf("\nThe poped value is %d",value);
+            s->top--;
+        }
+}
+/* Main function */
+void main()
+{
```
Comment (KanchanThareja) [inline Stacks_Using_Linked_Lists/Stacks.c]: @divya1999ann Also add suggested functions as in above comments.

**Comment by somya-kapoor** [comment]
@divya1999ann After doing all the requested changes, please squash all the commits.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 15]
```diff
@@ -27,13 +29,52 @@ void POP(struct stack *s)
...
+        printf("\nThe poped value is %d",value);
+        s->top--;
+    }
+    count--;
 }
+
+/* function to check if the stack is empty or not */
+int isEmpty() 
+{ 
+    return s->top == NULL;
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: error: 's' undeclared (first use in this function)

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 16]
```diff
@@ -27,13 +29,52 @@ void POP(struct stack *s)
...
+int isEmpty() 
+{ 
+    return s->top == NULL; 
+} 
+  
+/* function to return top element in a stack */ 
+int peek() 
+{ 
+    //check for empty stack 
+    if (!isEmpty(s->top))
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: error: 's' undeclared (first use in this function)

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 17]
```diff
@@ -27,13 +29,52 @@ void POP(struct stack *s)
...
+} 
+  
+/* function to return top element in a stack */ 
+int peek() 
+{ 
+    //check for empty stack 
+    if (!isEmpty(s->top)) 
+        return s->data[s->top]; 
+    else
+        exit(EXIT_FAILURE);
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: error: 'EXIT_FAILURE' undeclared (first use in this function)

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 18]
```diff
@@ -27,13 +29,52 @@ void POP(struct stack *s)
...
+{
+    if(Stack.top == -1)
+    {
+        printf("\nStack is empty");
+    }
+    else
+    {
+        printf("\nThe current stack is ");
+        for(int i = 0; i <= Stack.top; i++)
+        printf("%d\t",s->data[i]);
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: error: 's' undeclared (first use in this function)

File: Stacks_Using_Linked_Lists/Stacks.c (hunk 17 above)
Comment (divya1999ann) [reply Stacks_Using_Linked_Lists/Stacks.c]: EXIT_FAILURE macros is to indicate program execution status.It indicates an unsuccessful execution of a program

**Comment by divya1999ann** [comment]
@SakshiShreya I made the changes.Please verify the code.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 19]
```diff
@@ -0,0 +1,155 @@
+/* C program to implement stack using linked list*/
+
+#include<stdio.h>
+struct stack
+{
+int top;
+int data[100];
+}Stack;
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: Fix indentation of this struct.

File: Stacks_Using_Linked_Lists/Stacks.c (hunk 17 above)
Comment (SakshiShreya) [reply Stacks_Using_Linked_Lists/Stacks.c]: In which header file is this defined? I am getting error.

File: Stacks_Using_Linked_Lists/Stacks.c (hunk 19 above)
Comment (divya1999ann) [reply Stacks_Using_Linked_Lists/Stacks.c]: ok done

**Comment by divya1999ann** [comment]
I have made changes @SakshiShreya

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 20]
```diff
@@ -0,0 +1,155 @@
...
+    {
+        printf("\nWhich operation to perform 1.PUSH 2.POP 3.Check empty or not 4.Print top element 5.Display stack 6.Size");	 /* Menu Driven Programming */
+        printf("\nEnter choice");
+        scanf("%d",&c);
+        if(c == 1)
+        {
+            printf("\nEnter value to push ");
+            scanf("%d",&value);
+            PUSH(s,value,n);
+	    break;
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: Please check this if else block. Many lines are indented with tab.

Comment (divya1999ann) [reply Stacks_Using_Linked_Lists/Stacks.c]: ok have made changes @SakshiShreya

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 21]
```diff
@@ -0,0 +1,155 @@
...
+	    size(s);
+	    break;
+	}
+	else
+	{
+            break;
+	}
+        printf("\nDo u want to continue(1/0)");
+        scanf("%d",&k);
+    }while(k != 0);
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Fix space before while

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 22]
```diff
@@ -0,0 +1,155 @@
+/* C program to implement stack using linked list*/
+
+#include<stdio.h>
+struct stack
+{
+    int top;
+    int data[100];
+}Stack;
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Fix space before ``Stack``

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 23]
```diff
@@ -89,49 +90,49 @@ void main()
...
 	else
-	{
+        {
             break;
-	}
+        }
         printf("\nDo u want to continue(1/0)");
         scanf("%d",&k);
-    }while(k != 0);
+    } while (k != 0);
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: I can still see some lines indented with a tab in this if else block. Please fix.

Comment (divya1999ann) [reply Stacks_Using_Linked_Lists/Stacks.c]: @SakshiShreya I have made changes.

Comment (SakshiShreya) [reply Stacks_Using_Linked_Lists/Stacks.c]: Not fixed yet. These are some lines: 98, 102, 103 ...

**Comment by KanchanThareja** [comment]
Hey @divya1999ann I think you are doing this manually, indentation can be easily corrected with text editors.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 24]
```diff
@@ -0,0 +1,155 @@
...
+            size(s);
+            break;
+        }
+        else
+        {
+            break;
+        }
+        printf("\nDo u want to continue(1/0)");
+        scanf("%d",&k);
+    } while (k != 0);
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: The code does not run in loop. The loop breaks once we enter any condition in this if-else block.

Please fix this.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 25]
```diff
@@ -0,0 +1,155 @@
...
+{
+    struct stack *s;
+    s = &Stack;
+    Stack.top = -1;
+    int c, value, k, n;
+    printf("\nEnter the number of elements");
+    scanf("%d",&n);
+    do
+    {
+        printf("\nWhich operation to perform 1.PUSH 2.POP 3.Check empty or not 4.Print top element 5.Display stack 6.Size");	 /* Menu Driven Programming */
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: Please use `\n` to separate all the options in the menu. Right now it does not give a good reading experience.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 26]
```diff
@@ -0,0 +1,155 @@
...
+struct stack
+{
+    int top;
+    int data[100];
+} Stack; 
+
+int count = 0;
+
+// Function to push element to stack
+void PUSH(struct stack *s,int value,int n)
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: Use camel case to name all the functions.

**Comment by divya1999ann** [comment]
@SakshiShreya I have made changes.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 27]
```diff
@@ -0,0 +1,151 @@
+// C program to implement stack using linked list
+
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Remove this extra line (2).

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 28]
```diff
@@ -0,0 +1,151 @@
...
+struct stack
+{
+    int top;
+    int data[100];
+} Stack; 
+
+int count = 0;
+
+// Function to push element to stack
+void push(struct stack *s,int value,int n)
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Add space after each comma.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 29]
```diff
@@ -0,0 +1,151 @@
...
+// Function to pop element from stack 
+void pop(struct stack *s)
+{
+    if (s -> top == -1)
+        printf("\nThe stack is empty");
+    else
+    {
+        int value;
+        value = s -> data[s -> top];
+        printf("\nThe poped value is %d",value);
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Space after comma.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 30]
```diff
@@ -0,0 +1,151 @@
...
+{
+    if (Stack.top == -1)
+    {
+        printf("\nStack is empty");
+    }
+    else
+    {
+        printf("\nThe current stack is ");
+        for (int i = 0; i <= Stack.top; i++)
+        printf("%d\t",s -> data[i]);
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Space after comma.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 31]
```diff
@@ -0,0 +1,151 @@
...
+    }
+}
+
+// Function to print size of stack
+void size(struct stack *s)
+{
+    printf("\nSize of stack is %d",count);
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Fix spacing

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 32]
```diff
@@ -0,0 +1,151 @@
...
+
+// Main function
+void main()
+{
+    struct stack *s;
+    s = &Stack;
+    Stack.top = -1;
+    int c, value, k, n;
+    printf("\nEnter the number of elements");
+    scanf("%d",&n);
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Fix spacing

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 33]
```diff
@@ -0,0 +1,151 @@
...
+    do
+    {
+        printf("\nWhich operation to perform \n 1.Push \n 2.Pop \n 3.Check empty or not \n 4.Print top element \n 5.Display stack \n 6.Size");	 // Menu Driven Programming
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Break this long statement into 2-3 lines

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 34]
```diff
@@ -0,0 +1,151 @@
...
+        printf("\nEnter choice");
+        scanf("%d",&c);
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Fix spacing

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 35]
```diff
@@ -0,0 +1,151 @@
...
+        if (c == 1)
+        {
+            printf("\nEnter value to push ");
+            scanf("%d",&value);
+            push(s,value,n);
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Spacing in above two lines

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 36]
```diff
@@ -0,0 +1,151 @@
...
+        else if (c == 5)
+        {
+            printStack(s);
+        }
+        else if (c == 6)
+        {
+            size(s);
+        }
+        printf("\nDo u want to continue(1/0)");
+        scanf("%d",&k);
```
Comment (marwahmanbir) [inline Stacks_Using_Linked_Lists/Stacks.c]: Spacing

File: Stacks_Using_Linked_Lists/Stacks.c (hunk 24 above)
Comment (marwahmanbir) [reply Stacks_Using_Linked_Lists/Stacks.c]: @divya1999ann Is this ^ resolved?

File: Stacks_Using_Linked_Lists/Stacks.c (hunk 35 above)
Comment (divya1999ann) [reply Stacks_Using_Linked_Lists/Stacks.c]: can u show me an example for this.

**Comment by marwahmanbir** [comment]
@divya1999ann Is the loop issue resolved?

**Comment by divya1999ann** [comment]
> @divya1999ann Is the loop issue resolved?

yes it is solved

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 37]
```diff
@@ -84,54 +83,45 @@ void main()
     Stack.top = -1;
     int c, value, k, n;
     printf("\nEnter the number of elements");
-    scanf("%d",&n);
+    scanf("%d", &n);
     do
     {
-        printf("\nWhich operation to perform 1.PUSH 2.POP 3.Check empty or not 4.Print top element 5.Display stack 6.Size");	 /* Menu Driven Programming */
+        printf("\nWhich operation to perform \n 1.Push \n 2.Pop \n 3.Check empty or not);
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: Close the double quotes in line 89.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 38]
```diff
@@ -0,0 +1,151 @@
...
+5.Display stack 
+6.Size
+Enter choice 1
+Enter the value to push 3
+Do u want to continue(1/0) 1
+Which operation to perform 1.PUSH 2.POP 3.Check empty or not 4.Print top element 5.Display stack 6.Size
+Enter choice 2
+The popped value is 3
+Do u want to continue(1/0) 1
+Which operation to perform 1.PUSH 2.POP 3.Check empty or not 4.Print top element 5.Display stack 6.Size
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: Fix line 143 and 147. They should be similar to lines 133 to 139.

File: Stacks_Using_Linked_Lists/Stacks.c [hunk 39]
```diff
@@ -63,9 +63,9 @@ void printStack(struct stack *s)
     }
     else
     {
-        printf("\nThe current stack is ");
+        printf("\nThe current stack is\n");
         for (int i = 0; i <= Stack.top; i++)
-        printf("%d\t", s -> data[i]);
+        printf("%d\n", s -> data[i]);
```
Comment (SakshiShreya) [inline Stacks_Using_Linked_Lists/Stacks.c]: There was no need to add new line after each element. I just suggested to add a new line after 'The current stack is'.

**Pull request merged**